stdlib/stdlib_docs.rs

```diff
@@ -1,125 +1,80 @@
-    }
+fn markdown_file_name(ns: &str) -> String {
+    format!("{}_{}.md", get_module_name(ns), get_module_section(ns))
 }
 
-/// Holds the stdlib modules and corresponding functions.
-pub struct Stdlib {
-    modules: BTreeMap<String, Module>,
-}
-
-impl Stdlib {
-    pub fn new() -> Self {
-        Stdlib {
-            modules: BTreeMap::new(),
-        }
-    }
-}
-
-/// A renderer renders a stdlib struct into a particular doc format and index (e.g: markdown, etc)
+/// A renderer renders a ModuleMap into a particular doc format and index (e.g: markdown, etc)
 trait Renderer {
     // Render writes out the document files into the output directory
-    fn render(stdlib: &Stdlib, output_dir: &str);
+    fn render(stdlib: &ModuleMap, output_dir: &str);
 }
 
 struct MarkdownRenderer {}
 
 impl Renderer for MarkdownRenderer {
-    fn render(stdlib: &Stdlib, output_dir: &str) {
+    fn render(stdlib: &ModuleMap, output_dir: &str) {
         // Write per module markdown file
-        for (ns, module) in &stdlib.modules {
-            let file_name = module.markdown_file_name();
+        for (ns, module) in stdlib.iter() {
+            let file_name = markdown_file_name(ns);
             let file_path = Path::new(output_dir).join(file_name);
             let mut f = fs::OpenOptions::new()
                 .write(true)
                 .append(true)
                 .create(true)
                 .open(file_path)
                 .expect("unable to open stdlib markdown file");
-
-            // Render modules into markdown
-            f.write_all(module.comments.join("\n").as_bytes())
-                .expect("unable to write module comments");
             let header = format!(
                 "\n## {}\n| Procedure | Description |\n| ----------- | ------------- |\n",
                 ns
             );
 
             f.write_all(header.as_bytes())
                 .expect("unable to write header to writer");
-            for func in module.functions.iter() {
-                let func_output = format!(
-                    "| {} | {} |\n",
-                    func.name,
-                    func.comments.join("<br />").replace('|', "\\|")
-                );
-                f.write_all(func_output.as_bytes())
-                    .expect("unable to write func to writer");
+            for proc in module.local_procs.iter() {
+                if proc.docs.is_some() {
+                    let func_full_name = format!("{}.{}", proc.name, proc.num_locals);
+                    let func_output = format!(
+                        "| {} | {} |\n",
+                        func_full_name,
+                        proc.docs
+                            .clone()
+                            .unwrap()
+                            .replace('|', "\\|")
+                            .replace('\n', "<br /><br />")
+                    );
+                    f.write_all(func_output.as_bytes())
+                        .expect("unable to write func to writer");
```

Comment:
We should probably put this logic into a helper function. This function could take a reference to a write and a `Procedure` struct and write the procedure's doc comments into the writer.